mimecast/client.go

```diff
@@ -86,32 +90,61 @@ type AuditLog struct {
 }
 
 type MimecastConfig struct {
-	ClientOptions uspclient.ClientOptions `json:"client_options" yaml:"client_options"`
-	ClientId      string                  `json:"client_id" yaml:"client_id"`
-	ClientSecret  string                  `json:"client_secret" yaml:"client_secret"`
+	ClientOptions        uspclient.ClientOptions `json:"client_options" yaml:"client_options"`
+	ClientId             string                  `json:"client_id" yaml:"client_id"`
+	ClientSecret         string                  `json:"client_secret" yaml:"client_secret"`
+	BaseURL              string                  `json:"base_url,omitempty" yaml:"base_url,omitempty"`
+	InitialLookback      time.Duration           `json:"initial_lookback,omitempty" yaml:"initial_lookback,omitempty"` // eg, 24h, 30m, 168h, 1h30m
+	MaxConcurrentWorkers int                     `json:"max_concurrent_workers,omitempty" yaml:"max_concurrent_workers,omitempty"`
 }
 
 func (c *MimecastConfig) Validate() error {
 	if err := c.ClientOptions.Validate(); err != nil {
-		return fmt.Errorf("client_options: %v", err)
+		return fmt.Errorf("client_options: %v", err) // Required
 	}
+
 	if c.ClientId == "" {
-		return errors.New("missing client id")
+		return errors.New("missing client id") // Required
 	}
 	if c.ClientSecret == "" {
-		return errors.New("missing client secret")
+		return errors.New("missing client secret") // Required
+	}
+
+	if c.BaseURL == "" {
+		c.BaseURL = "https://api.services.mimecast.com" // Default Global - May process data in other regions during failover
+		// US Specific - Ensures data stays within the US instance: https://us-api.services.mimecast.com
+		// UK Specific - Ensures data stays within the UK instance: https://uk-api.services.mimecast.com
+	}
+
+	if c.InitialLookback == 0 {
+		c.InitialLookback = 24 * time.Hour // Default
+	} else {
+		if c.InitialLookback < 1*time.Minute {
+			return fmt.Errorf("initial_lookback must be at least 1 minute, got %s", c.InitialLookback)
+		}
+		if c.InitialLookback > 30*24*time.Hour { // 30 days
+			return fmt.Errorf("initial_lookback must be less than 30 days, got %s", c.InitialLookback)
+		}
+	}
```

Comment:
Adapters run ephemerally in the cloud, so they _could_ restart multiple times a day. So we should not have a min look back as that value becomes potentially redundant overlap. The look back is mostly useful for manual backfill jobs.
Similarly, if a user wants to do a backfill of 6 months, that should be something we allow.

So my suggestion is to remove the validations here.
```suggestion
```